go/arrow/internal/debug/assert_off.go

```diff
@@ -14,6 +14,7 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+//go:build !assert
```

Comment:
I don't know too much about these, https://stackoverflow.com/questions/68360688/whats-the-difference-between-gobuild-and-build-directives

I'm happy to run go-fix as well

Reply:
This is fine, it's because the format changed (in a backwards compatible way) around go 1.18 or so. `//go:build` is the correct current one for the latest versions of the Go toolchain